spec/jobs/notifications/post_created_job_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+describe Notifications::PostCreatedJob do
+  subject { described_class.new(actor_id, post_id) }
+  let(:topic) { create :topic }
+  let(:post) { create :post, topic: topic }
+  let(:coach) { create :faculty }
+  let(:actor_id) { post.creator.id }
+  let(:post_id) { post.id }
+
+  describe '#perform' do
+    before do
+      create :topic_subscription, topic: topic, user: coach.user
+    end
+
+    describe 'job scheduled with an in-valid post id' do
+      let(:post_id) { 'xxxx' }
+
+      it 'will not create notifications' do
+        subject.perform_now
+        expect(Notification.count).to eq(0)
+      end
+    end
+
+    describe 'Job scheduled with an in-valid actor id' do
+      let(:actor_id) { 'xxxx' }
+
+      it 'will not create notifications' do
+        subject.perform_now
+        expect(Notification.count).to eq(0)
+      end
+    end
+
+    scenario 'creates notification' do
```

Comment:
1. This isn't a `scenario`. It's an `it` - a simple, single assertion. A `scenario` will include a statement that describes a more complex user interaction.
2. I'd extract the valid actor and post IDs from the top-level and move it into a context here, since the parallel specs assume invalid ones.

```suggestion
    context 'with valid actor and post IDs' do
      let(:actor_id) { post.creator.id }
      let(:post_id) { post.id }
      
      it 'creates a notification' do
```